### PR TITLE
Expose CheckSuitesService and ChuckRunsService classes

### PR DESCRIPTION
**Pull request by kevmoo**
Also a bunch of other cleanup, enabled latest lints
removed duplicate lints
sorted lints
etc

### CODE REVIEW

**Comment by kevmoo** [comment]
Also takes care of https://github.com/SpinlockLabs/github.dart/pull/345

File: analysis_options.yaml [hunk 1]
```diff
@@ -1,397 +1,61 @@
 include: package:lints/recommended.yaml
 
 analyzer:
-  strong-mode:
```
Comment (CaseyHillers) [inline analysis_options.yaml:4]: Nice, thank you!

File: lib/src/common/checks_service.dart [hunk 2]
```diff
@@ -11,21 +11,21 @@ class ChecksService extends Service {
   /// Methods to interact with Check Runs.
   ///
   /// API docs: https://developer.github.com/v3/checks/runs/
-  final _CheckRunsService checkRuns;
+  final CheckRunsService checkRuns;
```
Comment (CaseyHillers) [inline lib/src/common/checks_service.dart:14]: I have some of my own code to cleanup now :-)

File: test/assets/responses/nocked_responses.dart [hunk 3]
```diff
@@ -1,4 +1,4 @@
-var getBlob = '''
+String getBlob = '''
```
Comment (CaseyHillers) [inline test/assets/responses/nocked_responses.dart:1]: Should these be final?

**Review by CaseyHillers — APPROVED** [review]
LGTM!

File: test/assets/responses/nocked_responses.dart (hunk 3 above)
Comment (kevmoo) [reply test/assets/responses/nocked_responses.dart:1]: or const?

**Pull request merged**